CONTRIBUTING.md

```diff
@@ -1,154 +1,171 @@
+follow our [beta testing guidance](https://github.com/flutter/devtools/blob/master/BETA_TESTING.md).
+
+## Welcome
+
+We gladly accept contributions via GitHub pull requests! We encourage you to read the
+[Welcome](https://github.com/flutter/flutter/blob/master/CONTRIBUTING.md#welcome) remarks in the Flutter
+framework's contributing guide, as all of that information applies to contributing to the `flutter/devtools`
+repo as well.
+
+We communicate primarily over GitHub and [Discord](https://github.com/flutter/flutter/wiki/Chat) on the
+[#hackers-devtools](https://discord.com/channels/608014603317936148/1106667330093723668) channel.
 
 Before contributing code:
 
 1. Complete the
 [Contributor License Agreement](https://cla.developers.google.com/clas).
 You can do this online, and it only takes a minute.
 
-2. Understand [coding agreements](packages/README.md).
-
-## Workflow for making changes
-
-- Change flutter to the latest flutter candidate:
-  `./tool/update_flutter_sdk.sh --local`
-- Create a branch from your cloned repo: `git checkout -b myBranch`
-- Ensure your branch, dependencies, and generated code are up to date: `devtools_tool sync`
-- Implement your changes
-- Commit work to your branch: `git commit -m “description”`
-- Push to your branch: `git push origin myBranch`
-- Navigate to the Pull Requests tab in the main [DevTools repo](https://github.com/flutter/devtools). You should see a popup to create a pull request from the branch in your cloned repo to DevTools master. Create a pull request.
+2. Review the [DevTools style guide](STYLE.md), which uses a combination of Dart and Flutter best practices.
+
+## Table of contents
+1. [Developing for DevTools](#developing-for-devtools)
+	- [Set up your DevTools environment](#set-up-your-devtools-environment)
+	- [Workflow for making changes](#workflow-for-making-changes)
+	- [Keeping your fork in sync](#keeping-your-fork-in-sync)
+2. [Running and debugging DevTools](#running-and-debugging-devtools)
+    - [Frontend only (most common)](#frontend-only-most-common)
+    - [Frontend + DevTools server](#frontend--devtools-server)
+    - [DevTools + VS Code integration](#devtools--vs-code-integration-ide-embedded-devtools-experience)
+3. [Testing for DevTools](#testing-for-devtools)
+4. [Appendix](#appendix)
+
+## Developing for DevTools
+
+### Set up your DevTools environment
+
+> Before setting up your DevTools environment, please make sure you have
+[cloned the Flutter SDK from GitHub](https://docs.flutter.dev/get-started/install/macos#downloading-straight-from-github-instead-of-using-an-archive)
+and added the included `flutter` and `dart` executables to your `PATH` environment variable (see Flutter
+instructions for how to [update your PATH](https://docs.flutter.dev/get-started/install/macos#update-your-path)).
+>
+> Typing `which flutter` and `which dart` into your terminal should print the path to those binaries from
```

Comment:
This only works on MacOS and Linux. Much of this documentation points to MacOS docs, but the majority of the community works on Windows. We should either have a separate section on getting setup on Windows or include Windows specific details inline.

Reply:
Added instructions for `where.exe flutter` and `where.exe dart`. @DanTup can you confirm?

Reply:
Yep, this looks correct. The `.exe` is unfortunately required in PowerShell because `where` is an alias for `Where-Object` (it's a shame `which` isn't an alias for `where.exe` 😄 )

![image](https://github.com/flutter/devtools/assets/1078012/05697e30-9f5f-4bcb-8af9-b9e1520191a3)